src/tests/utils.ts

```diff
@@ -127,3 +137,79 @@ export const resetLimitsForTesting = (): void => {
   bufferSize = null;
   jsonIndent = null;
 };
+
+export const transformTestResult = (rawResult: TestResultRaw): TestResult => {
+  // Destructure summary to omit testSetupTimeInMs
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { testSetupTimeInMs, ...summary } = rawResult.summary;
+
+  // Initialize arrays for setup methods and regular tests
+  const setupMethods: ApexTestResultData[] = [];
+  const regularTests: ApexTestResultData[] = [];
+
+  // Iterate through each item in rawResult.tests
+  rawResult.tests.forEach((test) => {
+    const { isTestSetup, ...rest } = test as ApexTestResultDataRaw;
+    (isTestSetup ? setupMethods : regularTests).push(rest);
+  });
+
+  return {
+    summary: {
+      ...summary,
+      testTotalTimeInMs:
+        (testSetupTimeInMs || 0) + summary.testExecutionTimeInMs
+    },
+    tests: regularTests,
+    setup: setupMethods,
+    codecoverage: rawResult.codecoverage
+  };
+};
+
+export const calculateCodeCoverage = async (
+  codeCoverageInstance: CodeCoverage,
+  codeCoverage: boolean,
+  apexTestClassIdSet: Set<string>,
+  result: TestResult,
+  coveredApexClassIdSet: Set<string>,
```

Comment:
I think this does not need to be passed in, look into making it local to the function.